fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -484,6 +484,8 @@ struct CommitBatchContext {
 
 	void setupTraceBatch();
 
+	std::set<Tag> writtenTags;
```

Comment:
This set includes replicas too, right? Dan also needs these tags in the version vector prototype.

Reply:
the code in this area is hard to follow. It appears to convert a tag to a list of locations (`msg_locations`), per the replica policy, which are indexes into the flat list of tlogs.  The long and short is we only know the replicas after `writeTypedMessage` is called. I pulled that information out of `msg_locations` array (and constructed a tag using it), maybe a similar trick can be used here. we can refactor this if the project makes it to master.

Reply:
>This set includes replicas too, right?
It does (per Jingyu) I will update my PR to reflect this.